BotRoyaleUnity/Assets/Scripts/RoombaMovement.cs

```diff
@@ -23,11 +22,22 @@ private IEnumerator ChangeDirectionCoroutine()
         // never stop this coroutine.
         while (true)
         {
-            // wait some time between changing directions
-            yield return new WaitForSeconds(ChangeDirectionAfterTime);
-
-            // change rotation to a random direction
-            transform.Rotate(Vector3.forward, Random.Range(0f, 360f));
+            float newDirection = Random.Range(-180f,180f);
+            float rotationSpeed = Random.Range(0.5f,1.5f);
+            float totalRotation = 0;
+            if(newDirection < 0){
+                while(totalRotation > newDirection){
+                    transform.Rotate(Vector3.forward, -rotationSpeed);
+                    totalRotation -= rotationSpeed;
```

Comment:
If you do the waiting to next frame, this will become framerate dependant, scaling it by time since last frame would make it framerate independant, something like
`var rotationThisFrame = rotationSpeed * Time.deltatime;
transform.Rotate(Vector3.forwared, rotationThisFrame);
`